src/policy/CMSRucioPolicy/algorithms/auto_approve.py

```diff
@@ -2,60 +2,43 @@
 Auto approve algorithm for CMS Rucio policy
 """
 import logging
-from sqlalchemy.sql import func
 from configparser import NoOptionError, NoSectionError
+from datetime import datetime
+
+from rucio.common.config import config_get
+from sqlalchemy.sql import func
 
 
-def global_approval(did, rule_attributes, session):
+def global_approval(rule, did, session) -> bool:
```

Comment:
@ericvaandering 
This change in the function signature is the main change. 
Instead of using attributes passed by the user, we take the rule as stored in the DB.